src/Internal/Wallet/Cip30Mock.purs

```diff
@@ -155,49 +211,69 @@ mkCip30Mock pKey mSKey = do
             Map.filter
               (flip Array.elem (collateralUtxos <#> unwrap >>> _.output))
               utxos
-        -- Convert to CSL representation and serialize
-        cslUtxos <- traverse (liftEffect <<< convertTransactionUnspentOutput)
-          $ Map.toUnfoldable nonCollateralUtxos <#> \(input /\ output) ->
-              TransactionUnspentOutput { input, output }
-        pure $ (byteArrayToHex <<< unwrap <<< toBytes) <$> cslUtxos
-    , getCollateral: fromAff do
+        let
+          xUtxos = Map.toUnfoldable nonCollateralUtxos <#> \(input /\ output) ->
+            TransactionUnspentOutput { input, output }
+        let
+          amountValue = DSV.convertValue =<< fromBytes =<< hexToCborBytes =<<
+            (uorToMaybe amount)
+        if (not hasEnoughAmount amountValue xUtxos) then pure $ maybeToNullOr
+          Nothing
+        else do
+          -- Convert to CSL representation and serialize
+          cslUtxos <- traverse (liftEffect <<< convertTransactionUnspentOutput)
+            xUtxos
+          page <- liftEffect
+            $ paginateArray (uorToMaybe pagination)
+            $
+              (cborBytesToHex <<< toBytes) <$> cslUtxos
+          pure $ maybeToNullOr $ Just page
+    , getCollateral: \{ amount } -> unsafePerformEffect $ fromAff do
         ownAddress <- (unwrap keyWallet).address config.networkId
         utxos <- liftMaybe (error "No UTxOs at address") =<<
           runQueryMInRuntime config runtime (utxosAt ownAddress)
         collateralUtxos <- getCollateralUtxos utxos
-        cslUnspentOutput <- liftEffect $ traverse
-          convertTransactionUnspentOutput
-          collateralUtxos
-        pure $ (byteArrayToHex <<< unwrap <<< toBytes) <$>
-          cslUnspentOutput
+        amountValue <- liftEffect $ convertAmount amount
+        if gt amountValue maxCollateralAmount then liftEffect $
+          raiseInvalidRequestError "Amount value is bigger than allowed 5 ADA"
+        else pure unit
+        if (not $ hasEnoughAmount (Just amountValue) collateralUtxos) then
+          liftEffect $ raiseInvalidRequestError
+            "Cannot find collateral to match required amount"
+        else do
+          cslUnspentOutput <- liftEffect $ traverse
+            convertTransactionUnspentOutput
+            collateralUtxos
+          pure $ (cborBytesToHex <<< toBytes) <$> cslUnspentOutput
```

Comment:
`when` can be used instead of `if then else` to reduce indentation

Reply:
How so?  If have two branches, while when has only one.

Reply:
Raising an error will cause the second clause to never be evaluated

Reply:
As far as I understand this style (not writing else, ) is not enforced by our style guide and I am myself stand against that style, so I will not change that.